test/backend/remote.js

```diff
@@ -52,10 +52,26 @@ function testNotFound(t, err) {
   }, 'Object contains no other fields')
 }
 
+// Helper function that conditionally tests whether
+// the server emits the failure event.
+//
+// Takes three arguments:
+//
+// 1. the test object (t)
+// 2. the server.on function (serverOn)
+// 3. the call site test count (testCount)
+function testFailureEvent(t, serverOn, testCount) {
+  if (serverOn) {
+    t.plan(testCount + 1)
+    serverOn('failure', t.pass.bind(t, 'The server emitted the failure event'))
+  }
+}
+
 // To run these tests from a new backend, create a DB instance, start a test server
 // and pass the config containing the connection params to this function. The tests
-// will run against that server.
-module.exports = function(cfg) {
+// will run against that server. Optional second argument is the bound server.on
+// function, to enable tests to listen for emitted server events.
```

Comment:
Only reason I made it optional was so that the tests wouldn't fail if anyone updates this repo without also having updated `fxa-auth-db-mem`. But maybe that's a feature.